datalad_next/processors/__init__.py

```diff
@@ -0,0 +1,33 @@
+""" Iterators that support subprocess pipelining and output processing
```

Comment:
I am wondering if `processors` is a good name, and whether this framing is optimal.

It is true that some iterators in here are working with iterables of `bytes`, but most are agnostic.

To me this is really generics, pretty much like `itertools`.

Reply:
I also wondered about the name. Changed it to `datalad_next.itertools` in ff8344485ea111661f960f95d331326a318df37b